lib/trivial.nix

```diff
@@ -205,6 +205,13 @@ rec {
   */
   inNixShell = builtins.getEnv "IN_NIX_SHELL" != "";
 
+  /* Determine whether the function is being called from inside pure-eval mode
+     by seeing whether `builtins` contains `currentSystem`. If not, we must be in
+     pure-eval mode. This also implies we are in a Flake.
```

Comment:
Pure evaluation mode doesn't imply flakes, though that's arguably not that common:
```
$ nix-instantiate --eval --pure-eval -E 'builtins ? currentSystem'
false
```